storage/tianmu/types/tianmu_num.cpp

```diff
@@ -58,8 +58,9 @@ TianmuNum &TianmuNum::Assign(int64_t value, short scale, bool is_double, common:
   this->is_double_ = is_double;
   this->attr_type_ = attrt;
 
-  if (scale != -1 && !is_double_) {
-    if (scale != 0 || attrt == common::ColumnType::UNK) {
+  if (scale != -1 &&
+      !is_double_) {  // check if construct decimal, the UNK is used on temp_table.cpp: GetValueString(..)
+    if (scale != 0 || attrt == common::ColumnType::UNK || attrt == common::ColumnType::BIT) {
```

Comment:
Why we do add condition of `attrt == common::ColumnType::BIT` here?

Reply:
> Why we do add condition of `attrt == common::ColumnType::BIT` here?

Emmm, I made a mistake here, the condition should be:
```
if ((scale != 0 && attrt != common::ColumnType::BIT) || attrt == common::ColumnType::UNK) {
```
The condition is used to judge if this type is decimal or not, so when `scale != 0`, it may be `bit` or `decimal`.